test/integration/graql/reasoner/query/GenerativeOperationalIT.java

```diff
@@ -266,66 +264,90 @@ public void whenGeneralisingAttributes_SubsumptionRelationHoldsBetweenPairs(){
-                        .flatMap(p -> Stream.concat(Stream.of(p), fuzzer.apply(p, txCtx)))
-                        .forEach(fuzzedPattern -> {
-                            ReasonerQueryImpl cQuery = reasonerQueryFactory.create(conjunction(fuzzedPattern));
-
-                            if (pQuery.isAtomic() && cQuery.isAtomic()) {
-                                ReasonerAtomicQuery queryA = (ReasonerAtomicQuery) pQuery;
-                                ReasonerAtomicQuery queryB = (ReasonerAtomicQuery) cQuery;
-                                QueryTestUtil.unification(queryA, queryB,true, UnifierType.STRUCTURAL);
-                                QueryTestUtil.unification(queryA, queryB,true, UnifierType.STRUCTURAL_SUBSUMPTIVE);
-                            }
-                        });
+    public void whenFuzzyingIdsWithBindingsPreserved_StructuralEquivalenceIsNotAffected() throws ExecutionException, InterruptedException {
+        Function<List<Pattern>, Supplier<Void>> func = (patterns) -> () -> {
+            try (Transaction tx = genericSchemaSession.transaction(Transaction.Type.READ)) {
+                TestTransactionProvider.TestTransaction testTx = (TestTransactionProvider.TestTransaction) tx;
+                ReasonerQueryFactory reasonerQueryFactory = testTx.reasonerQueryFactory();
+                TransactionContext txCtx = new TransactionContext(tx);
+
+                Operator fuzzer = Operators.fuzzIds();
+                for(Pattern pattern : patterns){
+                    ReasonerQueryImpl pQuery = reasonerQueryFactory.create(conjunction(pattern));
+                    Stream.concat(Stream.of(pattern), fuzzer.apply(pattern, txCtx))
+                            .flatMap(p -> Stream.concat(Stream.of(p), fuzzer.apply(p, txCtx)))
+                            .forEach(fuzzedPattern -> {
+                                ReasonerQueryImpl cQuery = reasonerQueryFactory.create(conjunction(fuzzedPattern));
+
+                                if (pQuery.isAtomic() && cQuery.isAtomic()) {
+                                    ReasonerAtomicQuery queryA = (ReasonerAtomicQuery) pQuery;
+                                    ReasonerAtomicQuery queryB = (ReasonerAtomicQuery) cQuery;
+                                    QueryTestUtil.unification(queryA, queryB,true, UnifierType.STRUCTURAL);
+                                    QueryTestUtil.unification(queryA, queryB,true, UnifierType.STRUCTURAL_SUBSUMPTIVE);
+                                }
+                            });
                 }
             }
+            return null;
+        };
+        processPatterns(new ArrayList<>(binaryRelationPatternTree.keySet()), func, 1);
+    }
+
+    private void processPatterns(List<Pattern> patterns, Function<List<Pattern>, Supplier<Void>> func, int threads) throws ExecutionException, InterruptedException {
+        ExecutorService executorService = Executors.newFixedThreadPool(4);
```

Comment:
will it be super slow if we just do one pattern per tx (tx will be super fast in 2.0) - then no risk of cache issues, and this code will be super simple:
```
patterns.parallelStream().map(...)
```

Reply:
Bit slower, nothing is executed here so there isn't much that is cached. Anyway, changing to simpler.